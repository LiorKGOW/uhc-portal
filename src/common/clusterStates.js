const clusterStates = {
  PENDING: 'pending',
  INSTALLING: 'installing',
  ERROR: 'error',
  READY: 'ready',
  UNINSTALLING: 'uninstalling',
  PATCHING: 'patching',
  UNKNOWN: 'unknown',
};

export default clusterStates;
