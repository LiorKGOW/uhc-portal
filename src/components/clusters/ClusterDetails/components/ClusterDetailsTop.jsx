import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';

import { LinkContainer } from 'react-router-bootstrap';
import { Spinner } from '@redhat-cloud-services/frontend-components';
import {
  Breadcrumb, BreadcrumbItem, Button, Alert, Split, SplitItem, Title,
} from '@patternfly/react-core';

import clusterStates from '../../common/clusterStates';
import ClusterActionsDropdown from '../../common/ClusterActionsDropdown';
import RefreshButton from '../../../common/RefreshButton/RefreshButton';
import ErrorTriangle from '../../common/ErrorTriangle';
import getClusterName from '../../../../common/getClusterName';
import { subscriptionStatuses } from '../../../../common/subscriptionTypes';
import ExpirationAlert from './ExpirationAlert';

function ClusterDetailsTop(props) {
  const {
    cluster,
    openModal,
    pending,
    refreshFunc,
    clusterIdentityProviders,
    organization,
    error,
    errorMessage,
    children,
  } = props;

  const clusterName = getClusterName(cluster);
  const hasIdentityProviders = clusterIdentityProviders.clusterIDPList.length > 0;
  const showIDPMessage = (cluster.managed
                          && cluster.state === clusterStates.READY
                          && !hasIdentityProviders);

  const isArchived = get(cluster, 'subscription.status', false) === subscriptionStatuses.ARCHIVED;

  const openIDPModal = () => {
    openModal('create-identity-provider');
  };

  const IdentityProvidersHint = () => (
    <Alert
      id="idpHint"
      variant="warning"
      isInline
      title="Missing Identity Providers"
    >
      Identity Providers determine how users log into the cluster.
      {' '}
      <Button variant="link" isInline onClick={openIDPModal}>Add OAuth Configuration</Button>
      {' '}
      to allow  others to log in.
    </Alert>
  );

  const consoleURL = cluster.console ? cluster.console.url : false;

  let launchConsole;
  if (consoleURL && (cluster.state !== clusterStates.UNINSTALLING)) {
    launchConsole = (
      <a href={consoleURL} target="_blank" rel="noopener noreferrer" className="pull-left">
        <Button variant="primary">Launch Console</Button>
      </a>
    );
  } else if (cluster.managed) {
    launchConsole = (
      <Button variant="primary" isDisabled title={cluster.state === clusterStates.UNINSTALLING ? 'The cluster is being uninstalled' : 'Admin console is not yet available for this cluster'}>
      Launch Console
      </Button>
    );
  } else if (cluster.canEdit) {
    launchConsole = (<Button variant="primary" onClick={() => openModal('edit-console-url', cluster)}>Add Console URL</Button>);
  }

  const actions = (
    <ClusterActionsDropdown
      disabled={!cluster.canEdit && !cluster.canDelete}
      cluster={cluster}
      organization={organization.details}
      showConsoleButton={false}
    />
  );

  const isRefreshing = pending
      || organization.pending
      || clusterIdentityProviders.pending;


  return (
    <div id="cl-details-top" className="top-row">
      <Split>
        <SplitItem>
          <Breadcrumb>
            <LinkContainer to="">
              <BreadcrumbItem to="#">
                Clusters
              </BreadcrumbItem>
            </LinkContainer>
            {isArchived && (
              <LinkContainer to="/archived">
                <BreadcrumbItem to="/archived">
                  Archived clusters
                </BreadcrumbItem>
              </LinkContainer>
            )}
            <BreadcrumbItem isActive>
              {clusterName}
            </BreadcrumbItem>
          </Breadcrumb>
        </SplitItem>
      </Split>
      <Split id="cl-details-top-row">
        <SplitItem>
          <Title headingLevel="h1" size="4xl" className="vertical-align">{clusterName}</Title>
        </SplitItem>
        <SplitItem>
          { isRefreshing && <Spinner className="cluster-details-spinner" /> }
          { error && <ErrorTriangle errorMessage={errorMessage} className="cluster-details-warning" /> }
        </SplitItem>
        <SplitItem isFilled />
        <SplitItem>
          <span id="cl-details-btns">
            { !isArchived ? (
              <>
                {launchConsole}
                {actions}
              </>
            ) : (
              <Button
                variant="secondary"
                onClick={() => openModal('unarchive-cluster', {
                  subscriptionID: cluster.subscription ? cluster.subscription.id : '',
                  name: clusterName,
                })}
                isDisabled={!cluster.canEdit}
              >
                Unarchive
              </Button>
            )}
            <RefreshButton id="refresh" autoRefresh refreshFunc={refreshFunc} />
          </span>
        </SplitItem>
      </Split>
      {showIDPMessage && (
      <Split>
        <SplitItem isFilled>
          {!clusterIdentityProviders.pending && cluster.canEdit && <IdentityProvidersHint />}
        </SplitItem>
      </Split>
      )}
      {cluster.expiration_timestamp
      && (
      <ExpirationAlert
        expirationTimestamp={cluster.expiration_timestamp}
      />
      )}
      {children}
    </div>
  );
}

ClusterDetailsTop.propTypes = {
  cluster: PropTypes.object,
  openModal: PropTypes.func.isRequired,
  refreshFunc: PropTypes.func.isRequired,
  pending: PropTypes.bool.isRequired,
  clusterIdentityProviders: PropTypes.object.isRequired,
  organization: PropTypes.object.isRequired,
  error: PropTypes.bool,
  errorMessage: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.node,
    PropTypes.element,
  ]),
  children: PropTypes.any,
};

export default ClusterDetailsTop;
