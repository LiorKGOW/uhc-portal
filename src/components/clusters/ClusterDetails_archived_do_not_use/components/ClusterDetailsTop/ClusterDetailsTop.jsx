import React, { useState } from 'react';
import get from 'lodash/get';
import PropTypes from 'prop-types';

import { Alert, Button, Flex, Split, SplitItem, Title } from '@patternfly/react-core';
import { Spinner } from '@redhat-cloud-services/frontend-components/Spinner';

import getClusterName from '~/common/getClusterName';
import { goZeroTime2Null } from '~/common/helpers';
import isAssistedInstallSubscription, {
  isAvailableAssistedInstallCluster,
  isUninstalledAICluster,
} from '~/common/isAssistedInstallerCluster';
import { HAS_USER_DISMISSED_RECOMMENDED_OPERATORS_ALERT } from '~/common/localStorageConstants';
import { useNavigate } from '~/common/routing';
import { normalizedProducts } from '~/common/subscriptionTypes';
import { PreviewLabel } from '~/components/clusters/common/PreviewLabel';
import Breadcrumbs from '~/components/common/Breadcrumbs';
import ButtonWithTooltip from '~/components/common/ButtonWithTooltip';
import modals from '~/components/common/Modal/modals';
import RefreshButton from '~/components/common/RefreshButton/RefreshButton';
import { SubscriptionCommonFields } from '~/types/accounts_mgmt.v1';

import ClusterActionsDropdown from '../../../common/archived_do_not_use/ClusterActionsDropdown';
import clusterStates, {
  hasInflightEgressErrors,
  isHibernating,
  isOffline,
} from '../../../common/clusterStates';
import ErrorTriangle from '../../../common/ErrorTriangle';
import HibernatingClusterCard from '../../../common/HibernatingClusterCard/HibernatingClusterCard';
import { isExtenalAuthenicationActive } from '../../clusterDetailsHelper';
import { shouldShowLogs } from '../Overview/InstallationLogView';

import ClusterNonEditableAlert from './components/ClusterNonEditableAlert';
import ClusterProgressCard from './components/ClusterProgressCard';
import ClusterStatusMonitor from './components/ClusterStatusMonitor';
import ExpirationAlert from './components/ExpirationAlert';
import GcpOrgPolicyAlert from './components/GcpOrgPolicyAlert';
import LimitedSupportAlert from './components/LimitedSupportAlert';
import { RecommendedOperatorsAlert } from './components/RecommendedOperatorsAlert/RecommendedOperatorsAlert';
import SubscriptionCompliancy from './components/SubscriptionCompliancy';
import TermsAlert from './components/TermsAlert';
import TransferClusterOwnershipInfo from './components/TransferClusterOwnershipInfo';

const IdentityProvidersHint = () => {
  const navigate = useNavigate();
  return (
    <Alert
      id="idpHint"
      className="pf-v5-u-mt-md"
      isInline
      title="Create an identity provider to access cluster"
    >
      Identity providers determine how you can log into the cluster. You&apos;ll need to set this up
      so you can access your cluster{' '}
      <p>
        <Button
          variant="link"
          isInline
          onClick={() =>
            navigate({
              hash: '#accessControl',
            })
          }
        >
          Create identity provider
        </Button>{' '}
      </p>
    </Alert>
  );
};

function ClusterDetailsTop(props) {
  const {
    cluster,
    openModal,
    pending,
    refreshFunc,
    clickRefreshFunc,
    clusterIdentityProviders,
    organization,
    error,
    errorMessage,
    children,
    canSubscribeOCP,
    canTransferClusterOwnership,
    canHibernateCluster,
    autoRefreshEnabled,
    toggleSubscriptionReleased,
    showPreviewLabel,
    logs,
    openDrawer,
    closeDrawer,
    selectedCardTitle,
  } = props;

  const hasAlertBeenDismissed = localStorage.getItem(
    HAS_USER_DISMISSED_RECOMMENDED_OPERATORS_ALERT,
  );
  const isArchived =
    get(cluster, 'subscription.status', false) === SubscriptionCommonFields.status.ARCHIVED;
  const isDeprovisioned =
    get(cluster, 'subscription.status', false) === SubscriptionCommonFields.status.DEPROVISIONED;

  const [showRecommendedOperatorsAlert, setShowRecommendedOperatorsAlert] = useState(
    !hasAlertBeenDismissed && !isArchived && !isDeprovisioned,
  );

  let topCard = null;

  if (isHibernating(cluster)) {
    topCard = <HibernatingClusterCard cluster={cluster} openModal={openModal} />;
  } else if (
    cluster &&
    !isAssistedInstallSubscription(cluster.subscription) &&
    (shouldShowLogs(cluster) || hasInflightEgressErrors(cluster))
  ) {
    topCard = <ClusterProgressCard cluster={cluster} />;
  }

  const isProductOSDTrial =
    get(cluster, 'subscription.plan.type', '') === normalizedProducts.OSDTRIAL;
  const isProductOSDRHM =
    get(cluster, 'subscription.plan.type', '') === normalizedProducts.OSD &&
    get(cluster, 'subscription.cluster_billing_model', '') ===
      SubscriptionCommonFields.cluster_billing_model.MARKETPLACE;
  const isOSD = get(cluster, 'subscription.plan.type') === normalizedProducts.OSD;
  const isROSA = get(cluster, 'subscription.plan.type') === normalizedProducts.ROSA;
  const clusterName = getClusterName(cluster);
  const consoleURL = cluster.console ? cluster.console.url : false;
  const creationDateStr = get(cluster, 'creation_timestamp', '');

  const hasIdentityProviders = clusterIdentityProviders.clusterIDPList.length > 0;
  const showIDPMessage =
    cluster.managed &&
    cluster.idpActions?.create &&
    cluster.state === clusterStates.READY &&
    clusterIdentityProviders.fulfilled &&
    !hasIdentityProviders &&
    !isExtenalAuthenicationActive(cluster);

  const canUpgradeTrial = cluster.state === clusterStates.READY && cluster.canEdit;
  const trialExpirationUpgradeProps = canUpgradeTrial
    ? {
        trialExpiration: true,
        openModal,
        cluster,
      }
    : {};

  let launchConsole;
  if (consoleURL && !isOffline(cluster)) {
    launchConsole = (
      <a
        href={consoleURL}
        target="_blank"
        rel="noopener noreferrer"
        className="pull-left"
        data-testid="console-url-link"
      >
        <Button variant="primary">Open console</Button>
      </a>
    );
  } else if (cluster.managed) {
    launchConsole = (
      <Button
        variant="primary"
        isDisabled
        title={
          cluster.state === clusterStates.UNINSTALLING
            ? 'The cluster is being uninstalled'
            : 'Admin console is not yet available for this cluster'
        }
      >
        Open console
      </Button>
    );
  } else if (cluster.canEdit && !isUninstalledAICluster(cluster)) {
    launchConsole = (
      <Button variant="primary" onClick={() => openModal(modals.EDIT_CONSOLE_URL, cluster)}>
        Add console URL
      </Button>
    );
  }

  const actions = !isUninstalledAICluster(cluster) && (
    <ClusterActionsDropdown
      disabled={!cluster.canEdit && !cluster.canDelete}
      cluster={cluster}
      organization={organization.details}
      showConsoleButton={false}
      canSubscribeOCP={canSubscribeOCP}
      canTransferClusterOwnership={canTransferClusterOwnership}
      toggleSubscriptionReleased={toggleSubscriptionReleased}
      canHibernateCluster={canHibernateCluster}
      refreshFunc={refreshFunc}
    />
  );

  const breadcrumbs = (
    <Breadcrumbs
      path={[
        { label: 'Cluster List' },
        (isArchived || isDeprovisioned) && { label: 'Cluster Archives', path: '/archived' },
        { label: clusterName },
      ].filter(Boolean)}
    />
  );

  const isRefreshing = pending || organization.pending || clusterIdentityProviders.pending;

  const trialEndDate = isProductOSDTrial && get(cluster, 'subscription.trial_end_date');
  const OSDRHMEndDate =
    isProductOSDRHM && goZeroTime2Null(get(cluster, 'subscription.billing_expiration_date'));

  const canNotEditReason =
    !cluster.canEdit && 'You do not have permissions to unarchive this cluster';

  const unarchiveBtn = (
    <ButtonWithTooltip
      variant="secondary"
      onClick={() => {
        openModal(modals.UNARCHIVE_CLUSTER, {
          subscriptionID: cluster.subscription ? cluster.subscription.id : '',
          name: clusterName,
        });
      }}
      disableReason={canNotEditReason}
    >
      Unarchive
    </ButtonWithTooltip>
  );

  const orgPolicyWarning = logs?.find(
    (obj) =>
      obj.summary?.includes('Please enable the Org Policy API for the GCP project') ||
      obj.summary?.includes('GCP Organization Policy Service'),
  );

  const showGcpOrgPolicyWarning =
    orgPolicyWarning &&
    !isDeprovisioned &&
    cluster.state !== clusterStates.READY &&
    cluster.state !== clusterStates.UNINSTALLING;

  const shouldShowStatusMonitor =
    [
      clusterStates.WAITING,
      clusterStates.PENDING,
      clusterStates.INSTALLING,
      clusterStates.ERROR,
      clusterStates.UNINSTALLING,
    ].includes(cluster.state) || hasInflightEgressErrors(cluster);

  return (
    <div id="cl-details-top" className="top-row">
      <Split>
        <SplitItem className="pf-v5-u-pb-md">{breadcrumbs}</SplitItem>
      </Split>
      <Split id="cl-details-top-row">
        <SplitItem>
          <Title size="2xl" headingLevel="h1" className="cl-details-page-title">
            {clusterName}
            {showPreviewLabel && (
              <PreviewLabel creationDateStr={creationDateStr} className="pf-v5-u-ml-sm" />
            )}
          </Title>
        </SplitItem>
        <SplitItem>
          {isRefreshing && <Spinner className="cluster-details-spinner" />}
          {error && (
            <ErrorTriangle errorMessage={errorMessage} className="cluster-details-warning" />
          )}
        </SplitItem>
        <SplitItem isFilled />
        <SplitItem>
          <Flex
            flexWrap={{ default: 'nowrap' }}
            alignItems={{ default: 'alignItemsCenter' }}
            spaceItems={{ default: 'spaceItemsSm' }}
            id="cl-details-btns"
          >
            {!isArchived && !isDeprovisioned ? (
              <>
                {launchConsole}
                {actions}
              </>
            ) : (
              !isDeprovisioned && unarchiveBtn
            )}
            {!isDeprovisioned && !isArchived && (
              <RefreshButton
                id="refresh"
                autoRefresh={autoRefreshEnabled}
                refreshFunc={refreshFunc}
                clickRefreshFunc={clickRefreshFunc}
                useShortTimer={!Object.values(clusterStates).includes(cluster.state)}
              />
            )}
          </Flex>
        </SplitItem>
      </Split>

      {topCard}

      <LimitedSupportAlert
        limitedSupportReasons={cluster.limitedSupportReasons}
        isOSD={isOSD}
        isROSA={isROSA}
      />

      {shouldShowStatusMonitor ? (
        <ClusterStatusMonitor refresh={refreshFunc} cluster={cluster} />
      ) : null}

      {showIDPMessage && (
        <Split>
          <SplitItem isFilled>
            <IdentityProvidersHint />
          </SplitItem>
        </Split>
      )}
      {cluster.expiration_timestamp && (
        <ExpirationAlert expirationTimestamp={cluster.expiration_timestamp} />
      )}
      {trialEndDate && !isDeprovisioned && (
        <ExpirationAlert expirationTimestamp={trialEndDate} {...trialExpirationUpgradeProps} />
      )}
      {OSDRHMEndDate && !isDeprovisioned && (
        <ExpirationAlert expirationTimestamp={OSDRHMEndDate} OSDRHMExpiration />
      )}
      {showGcpOrgPolicyWarning && <GcpOrgPolicyAlert summary={orgPolicyWarning?.summary} />}

      <SubscriptionCompliancy
        cluster={cluster}
        openModal={openModal}
        canSubscribeOCP={canSubscribeOCP}
      />
      {!cluster.canEdit && isAvailableAssistedInstallCluster(cluster) && (
        <ClusterNonEditableAlert />
      )}
      <TransferClusterOwnershipInfo subscription={cluster.subscription} />
      <TermsAlert subscription={cluster.subscription} />
      {showRecommendedOperatorsAlert ? (
        <RecommendedOperatorsAlert
          openLearnMore={openDrawer}
          selectedCardTitle={selectedCardTitle}
          closeDrawer={closeDrawer}
          onDismissAlertCallback={() => setShowRecommendedOperatorsAlert(false)}
          clusterState={cluster.state}
          consoleURL={consoleURL}
        />
      ) : null}
      {children}
    </div>
  );
}

ClusterDetailsTop.propTypes = {
  cluster: PropTypes.object,
  openModal: PropTypes.func.isRequired,
  refreshFunc: PropTypes.func.isRequired,
  clickRefreshFunc: PropTypes.func,
  pending: PropTypes.bool.isRequired,
  clusterIdentityProviders: PropTypes.object.isRequired,
  organization: PropTypes.object.isRequired,
  error: PropTypes.bool,
  errorMessage: PropTypes.oneOfType([PropTypes.string, PropTypes.node, PropTypes.element]),
  children: PropTypes.any,
  canSubscribeOCP: PropTypes.bool.isRequired,
  canHibernateCluster: PropTypes.bool.isRequired,
  canTransferClusterOwnership: PropTypes.bool.isRequired,
  autoRefreshEnabled: PropTypes.bool,
  toggleSubscriptionReleased: PropTypes.func.isRequired,
  showPreviewLabel: PropTypes.bool.isRequired,
  logs: PropTypes.arrayOf(
    PropTypes.shape({
      summary: PropTypes.string,
      description: PropTypes.string,
    }),
  ),
  openDrawer: PropTypes.func,
  closeDrawer: PropTypes.func,
  selectedCardTitle: PropTypes.string,
};

export default ClusterDetailsTop;
