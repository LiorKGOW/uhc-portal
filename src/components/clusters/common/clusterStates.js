import get from 'lodash/get';
import { subscriptionStatuses } from '../../../common/subscriptionTypes';

const clusterStates = {
  PENDING: 'pending',
  INSTALLING: 'installing',
  UPDATING: 'updating',
  READY: 'ready',
  UNINSTALLING: 'uninstalling',
  HIBERNATING: 'hibernating',
  POWERING_DOWN: 'powering_down',
  RESUMING: 'resuming',
  DISCONNECTED: 'disconnected',
  ERROR: 'error',
  DEPROVISIONED: 'deprovisioned',
  ARCHIVED: 'archived',
};

function getClusterStateAndDescription(cluster) {
  let state;

  if ((cluster.state === clusterStates.INSTALLING
      || cluster.state === clusterStates.PENDING)) {
    state = clusterStates.INSTALLING;
  } else if (get(cluster, 'metrics.upgrade.state') === 'running' && !cluster.managed) {
    state = clusterStates.UPDATING;
  }
  if (!cluster.managed
    && cluster.subscription.status === subscriptionStatuses.DISCONNECTED) {
    state = clusterStates.DISCONNECTED;
  } else if (cluster.subscription.status === subscriptionStatuses.DEPROVISIONED) {
    state = clusterStates.DEPROVISIONED;
  } else if (cluster.subscription.status === subscriptionStatuses.ARCHIVED) {
    state = clusterStates.ARCHIVED;
  } else if (cluster.state === clusterStates.READY) {
    state = clusterStates.READY;
  } else if (cluster.state === clusterStates.UNINSTALLING) {
    state = clusterStates.UNINSTALLING;
  } else if (cluster.state === clusterStates.ERROR) {
    state = clusterStates.ERROR;
  } else if (cluster.state === clusterStates.HIBERNATING) {
    state = clusterStates.HIBERNATING;
  } else if (cluster.state === clusterStates.POWERING_DOWN) {
    state = clusterStates.POWERING_DOWN;
  } else if (cluster.state === clusterStates.RESUMING) {
    state = clusterStates.RESUMING;
  }


  return {
    state,
    // Capitalize the first letter and replace any underscore with space.
    description: state ? (state.charAt(0).toUpperCase() + state.slice(1)).replace(/_/g, ' ') : '',
  };
}

const isHibernating = state => state === clusterStates.HIBERNATING
  || state === clusterStates.POWERING_DOWN
  || state === clusterStates.RESUMING;

export { getClusterStateAndDescription, isHibernating };
export default clusterStates;
