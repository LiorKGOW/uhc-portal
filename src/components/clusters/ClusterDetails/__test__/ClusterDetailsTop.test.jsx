import React from 'react';
import { shallow } from 'enzyme';
// TODO: Remove this import when PF team fixes the issue causing tests to break without it
import { Button } from '@patternfly/react-core';

import ClusterDetailsTop from '../components/ClusterDetailsTop';
import fixtures, { funcs } from './ClusterDetails.fixtures';
import clusterStates from '../../common/clusterStates';

describe('<ClusterDetailsTop />', () => {
  let wrapper;
  const functions = funcs();

  beforeEach(() => {
    const props = {
      cluster: fixtures.clusterDetails.cluster,
      openModal: functions.openModal,
      pending: fixtures.clusterDetails.pending,
      refreshFunc: functions.refreshFunc,
      clusterIdentityProviders: fixtures.clusterIdentityProviders,
      organization: fixtures.organization,
      canSubscribeOCP: fixtures.canSubscribeOCP,
      canTransferClusterOwnership: fixtures.canTransferClusterOwnership,
      canHibernateCluster: fixtures.canHibernateCluster,
      toggleSubscriptionReleased: functions.toggleSubscriptionReleased,
    };
    wrapper = shallow(
      <ClusterDetailsTop {...props} />,
    );
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show refresh button', () => {
    expect(wrapper.find('RefreshBtn').length).toEqual(1);
  });

  it('should enable open console button when cluster has console url and cluster is not uninstalling', () => {
    const launchConsoleDisabled = wrapper.find(Button).at(0).props().isDisabled;
    expect(launchConsoleDisabled).toBeFalsy();
  });

  it('should disable open console button when console url is missing', () => {
    const cluster = { ...fixtures.clusterDetails.cluster, console: { url: '' } };
    wrapper.setProps({ cluster }, () => {
      const launchConsoleDisabled = wrapper.find(Button).at(0).props().isDisabled;
      expect(launchConsoleDisabled).toEqual(true);
    });
  });

  it('should disable open console button when cluster is unistalling', () => {
    const cluster = { ...fixtures.clusterDetails.cluster, state: clusterStates.UNINSTALLING };
    wrapper.setProps({ cluster }, () => {
      const launchConsoleDisabled = wrapper.find(Button).at(0).props().isDisabled;
      expect(launchConsoleDisabled).toEqual(true);
    });
  });

  it('should show error triangle if an error occured', () => {
    wrapper.setProps({ error: true }, () => {
      expect(wrapper.find('ErrorTriangle').length).toEqual(1);
    });
  });

  it('should show only Unarchive button if the cluster is archived', () => {
    const cluster = { ...fixtures.clusterDetails.cluster, subscription: { status: 'Archived', id: 'fake' } };
    wrapper.setProps({ cluster }, () => {
      const unarchiveButton = wrapper.find(Button).at(0);
      expect(unarchiveButton.props().variant).toEqual('secondary');
      expect(unarchiveButton.props().children).toEqual('Unarchive');
      expect(wrapper.find('ClusterActionsDropdown').length).toEqual(0); // no cluster actions dropdown
      expect(wrapper.find('RefreshBtn').length).toEqual(0); // no refresh button
      unarchiveButton.simulate('click');
      expect(functions.openModal).toBeCalledWith('unarchive-cluster', { subscriptionID: 'fake', name: cluster.name });
    });
  });
});
