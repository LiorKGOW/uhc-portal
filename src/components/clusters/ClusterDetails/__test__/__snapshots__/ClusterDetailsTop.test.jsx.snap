// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClusterDetailsTop /> should render 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={true}
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={8}
    >
      <Breadcrumb
        title={false}
      >
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/clusters"
        >
          <BreadcrumbItem
            active={false}
            href="#"
          >
            Clusters
          </BreadcrumbItem>
        </LinkContainer>
        <BreadcrumbItem
          active={true}
        >
          test-liza
        </BreadcrumbItem>
      </Breadcrumb>
    </Col>
  </Row>
  <hr />
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="cl-details-cluster-name"
      componentClass="div"
      sm={6}
    >
      <h1>
        <ClusterBadge
          clusterName="test-liza"
        />
      </h1>
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      lg={5}
      lgOffset={1}
    >
      <ButtonGroup
        block={false}
        bsClass="btn-group"
        id="cl-details-btns"
        justified={false}
        vertical={false}
      >
        <a
          className="pull-left"
          href="https://console-openshift-console.apps.test-liza.sdev.devshift.net"
          rel="noreferrer"
          target="_blank"
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="primary"
            disabled={false}
          >
            Launch Console
          </Button>
        </a>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          onClick={[Function]}
        >
          Admin Credentials
        </Button>
        <Connect(ClusterCredentialsModal)
          credentials={
            Object {
              "admin": Object {
                "password": "testPassword",
                "user": "kubeadmin",
              },
              "href": "/api/clusters_mgmt/v1/1IztzhAGrbjtKkMbiPewJanhTXk/credentials",
              "id": "1IztzhAGrbjtKkMbiPewJanhTXk",
              "kind": "ClusterCredentials",
              "kubeconfig": "apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: test",
              "ssh": Object {
                "private_key": "test",
                "public_key": "test",
              },
            }
          }
        />
        <DropdownButton
          bsStyle="default"
          disabled={false}
          id="actions"
          pullRight={true}
          title="Actions"
        >
          <Connect(ClusterActionsDropdown)
            cluster={
              Object {
                "api": Object {
                  "url": "https://api.test-liza.sdev.devshift.net:6443",
                },
                "canDelete": true,
                "canEdit": true,
                "cloud_provider": Object {
                  "href": "/api/clusters_mgmt/v1/cloud_providers/aws",
                  "id": "aws",
                  "kind": "CloudProviderLink",
                },
                "console": Object {
                  "url": "https://console-openshift-console.apps.test-liza.sdev.devshift.net",
                },
                "creation_timestamp": "2019-03-26T15:18:39.06783Z",
                "creator": "***REMOVED***",
                "display_name": "test-liza",
                "dns": Object {
                  "base_domain": "sdev.devshift.net",
                },
                "external_id": "9f50940b-fba8-4c59-9c6c-d64284b2026d",
                "flavour": Object {
                  "href": "/api/clusters_mgmt/v1/flavours/osd-4",
                  "id": "osd-4",
                  "kind": "FlavourLink",
                },
                "href": "/api/clusters_mgmt/v1/clusters/1IztzhAGrbjtKkMbiPewJanhTXk",
                "id": "1IztzhAGrbjtKkMbiPewJanhTXk",
                "kind": "Cluster",
                "managed": true,
                "metrics": Object {
                  "cpu": Object {
                    "total": Object {
                      "unit": "",
                      "value": 0,
                    },
                    "updated_timestamp": "0001-01-01T00:00:00Z",
                    "used": Object {
                      "unit": "",
                      "value": 0,
                    },
                  },
                  "memory": Object {
                    "total": Object {
                      "unit": "B",
                      "value": 0,
                    },
                    "updated_timestamp": "0001-01-01T00:00:00Z",
                    "used": Object {
                      "unit": "B",
                      "value": 0,
                    },
                  },
                  "nodes": Object {
                    "compute": 1,
                    "master": 2,
                    "total": 3,
                  },
                  "storage": Object {
                    "total": Object {
                      "unit": "B",
                      "value": 0,
                    },
                    "updated_timestamp": "0001-01-01T00:00:00Z",
                    "used": Object {
                      "unit": "B",
                      "value": 0,
                    },
                  },
                },
                "multi_az": false,
                "name": "test-liza",
                "network": Object {
                  "machine_cidr": "10.0.0.0/16",
                  "pod_cidr": "10.128.0.0/14",
                  "router_shards": Object {
                    "href": "/api/clusters_mgmt/v1/clusters/1IztzhAGrbjtKkMbiPewJanhTXk/router_shards",
                    "id": "1IztzhAGrbjtKkMbiPewJanhTXk",
                    "kind": "RouterShardListLink",
                  },
                  "service_cidr": "172.30.0.0/16",
                },
                "nodes": Object {
                  "compute": 2,
                  "master": 3,
                  "total": 5,
                },
                "region": Object {
                  "href": "/api/clusters_mgmt/v1/cloud_providers/aws/regions/us-east-1",
                  "id": "us-east-1",
                  "kind": "CloudRegionLink",
                },
                "state": "ready",
                "subscriptionInfo": Object {
                  "cluster_id": "1IztzhAGrbjtKkMbiPewJanhTXk",
                  "created_at": "2019-01-02T18:28:14.851121Z",
                  "external_cluster_id": "test-liza",
                  "href": "/api/accounts_mgmt/v1/subscriptions/1FDpnxsGxqFFFp2VNIWp5VajPc8",
                  "id": "1FDpnxsGxqFFFp2VNIWp5VajPc8",
                  "kind": "Subscription",
                  "last_telemetry_date": "0001-01-01T00:00:00Z",
                  "plan": Object {
                    "href": "/api/accounts_mgmt/v1/plans/OCP",
                    "id": "OCP",
                    "kind": "Plan",
                  },
                  "registry_credential": Object {
                    "href": "/api/accounts_mgmt/v1/registry_credentials/1EaZd2cDHH6ibIb1FFqav2Mles6",
                    "id": "1EaZd2cDHH6ibIb1FFqav2Mles6",
                    "kind": "RegistryCredential",
                  },
                  "updated_at": "2019-01-02T18:28:14.851121Z",
                },
                "version": Object {
                  "href": "/api/clusters_mgmt/v1/versions/openshift-v4.0-latest",
                  "id": "openshift-v4.0-latest",
                  "kind": "VersionLink",
                },
              }
            }
            hasIDP={false}
            idpID=""
            organization={
              Object {
                "created_at": "2019-01-02T18:28:14.851121Z",
                "ebs_account_id": "8888888",
                "external_id": "9999999",
                "href": "/api/accounts_mgmt/v1/organizations/1Iin1ev64UIFFjRPT5S8TfLr0Tj",
                "id": "1Iin1ev64UIFFjRPT5S8TfLr0Tj",
                "kind": "Organization",
                "name": "Test Corp",
                "updated_at": "2019-01-02T18:28:14.851121Z",
              }
            }
            showConsoleButton={false}
            showIDPButton={true}
          />
        </DropdownButton>
        <RefreshBtn
          autoRefresh={true}
          classOptions=""
          id="refresh"
          refreshFunc={[MockFunction]}
        />
      </ButtonGroup>
    </Col>
  </Row>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={12}
    >
      <IdentityProvidersHint />
    </Col>
  </Row>
</Grid>
`;
